### PR TITLE
Minor reorganization

### PR DESCRIPTION
**Pull request by jacobdadams**
I've reorganized the script to hopefully make it more self-documenting and easier to follow. I've not tried to give it a proper CLI or config file interface, just clean it up so that it's easier to understand when we come back to it after months of not using it. Extraneous functions and complicated logic have been replaced where possible.

### CODE REVIEW

File: FormatCountyCSV.py [hunk 1]
```diff
@@ -1,3 +1,7 @@
+"""
+OBSOLETE?
+"""
+
 import csv
 if __name__ == '__main__':
     countyFile = r'C:\GisWork\UspsCountyProject\CountyProjectOutputFile.txt'
```
Comment (steveoh) [inline FormatCountyCSV.py]: **nitpick**: this should be deleted

File: agrc_county_project.py [hunk 2]
```diff
@@ -154,107 +40,124 @@ def createCsvRow(row):
...
-            countyRow.append(addrRow[addrFields.index('SHAPE@Y')])
-            # County Project LONGITUDE
-            countyRow.append(addrRow[addrFields.index('SHAPE@X')])
-            # CountyProject row insert
-            countyProjectCursor.insertRow(countyRow)
-            # County project rows
-            counties[countyRow[0]].append(countyRow)
+    #: District identity datasets
+    congressional_districts_fc_path = Path(sgid_connection, 'SGID.POLITICAL.USCongressDistricts2012')
+    identify_result_fc_path = Path(output_folder, output_gdb_name, 'Addresses_Districts' + unique_run_id)
```
Comment (steveoh) [inline agrc_county_project.py]:
```suggestion
    identify_result_fc_path = Path(output_folder, output_gdb_name, f'Addresses_Districts{unique_run_id}')
```

File: agrc_county_project.py [hunk 3]
```diff
@@ -154,107 +40,124 @@ def createCsvRow(row):
...
-                        rowString = createCsvRow([str(x) for x in row])
-                        outputFile.write(rowString + '\n')
-                    except:
-                        print(row)
+    #: Make layer of only our specified counties
+    county_selection_where = None
+    if county_ids:
+        county_ids_string = ','.join(county_ids)
+        county_selection_where = f"CountyID IN ('{county_ids_string}')"
+    address_layer = 'addrpoints_' + unique_run_id
```
Comment (steveoh) [inline agrc_county_project.py]:
```suggestion
    address_layer = f'addrpoints_{unique_run_id}'
```

File: agrc_county_project.py [hunk 4]
```diff
@@ -154,107 +40,124 @@ def createCsvRow(row):
...
+    #: Get the congressional district for each address point
+    print('Running Congressional District identify...')
+    arcpy.env.outputCoordinateSystem = arcpy.SpatialReference(4326)
+    arcpy.env.geographicTransformations = 'NAD_1983_To_WGS_1984_5'
+    arcpy.Identity_analysis(address_layer, str(congressional_districts_fc_path), str(identify_result_fc_path))
+
+    #: Copy address points w/district info to dict of counties containing list of dicts of addresses
+    #: {county: [{address, city, etc}, ...], ...}
+    print('Loading and formatting source data...')
+    counties = defaultdict(list)  #: defaultdict adds new key with an empty list if key is not yet present
```
Comment (steveoh) [inline agrc_county_project.py:113]: **thought**: i think this is the same as using `counties.setdefault('key', []).append(item)`. no need to replace it just an fyi

**Review by steveoh — APPROVED** [review]
This would really benefit from a CLI to remove the need to edit file paths etc. It's a great improvement. The future person running it will appreciate this work.

File: agrc_county_project.py (hunk 4 above)
Comment (jacobdadams) [reply agrc_county_project.py:113]: That's my understanding as well. Some searching says that a defaultdict is faster than setdefault, but it probably doesn't matter here. I really just wanted an excused to try it :)

**Comment by jacobdadams** [comment]
The more I think about it, the more I think it actually makes more sense as a script tool/python toolbox than a cli script. There's so much work picking and validating feature classes and paths that would be annoying on the command line. Perhaps a later PR.

**Pull request merged**